src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
Sure, that's fine. Let me clean up the patch with that in mind and I'll ping you to have a look